server_test.go

```diff
@@ -308,3 +310,72 @@ func TestServer_testStdLogger(t *testing.T) {
 		t.Fatalf("expected: %q\ngot: %q", "test log", logOut.String())
 	}
 }
+
+func TestUnixSocketDir(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Unix sockets not supported on Windows")
+	}
+
+	tmpDir := t.TempDir()
+	t.Setenv(EnvUnixSocketDir, tmpDir)
+
+	closeCh := make(chan struct{})
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// make a server, but we don't need to attach to it
+	ch := make(chan *ReattachConfig, 1)
+	go Serve(&ServeConfig{
+		HandshakeConfig: testHandshake,
+		Plugins:         testGRPCPluginMap,
+		GRPCServer:      DefaultGRPCServer,
+		Logger:          hclog.NewNullLogger(),
+		Test: &ServeTestConfig{
+			Context:          ctx,
+			CloseCh:          closeCh,
+			ReattachConfigCh: ch,
+		},
+	})
+
+	// Wait for the server
+	var cfg *ReattachConfig
+	select {
+	case cfg = <-ch:
+		if cfg == nil {
+			t.Fatal("attach config should not be nil")
+		}
+	case <-time.After(2000 * time.Millisecond):
+		t.Fatal("should've received reattach")
+	}
+
+	actualDir := path.Clean(path.Dir(cfg.Addr.String()))
+	expectedDir := path.Clean(tmpDir)
+	if actualDir != expectedDir {
+		t.Fatalf("Expected socket in dir: %s, but was in %s", expectedDir, actualDir)
+	}
+
+	cancel()
+	<-closeCh
+}
+
+func TestUnixSocketGroupPermissions(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Unix sockets not supported on Windows")
```

Comment:
I am not a windows expert, but I don't think this is true anymore: https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/

Reply:
Wow TIL - I did have a little play around with letting these tests run on Windows, but Windows doesn't support Chown or Chmod in the same way, and go-plugin never creates Unix sockets on Windows anyway, so I'll still omit the tests for now as I was getting a weird failure.

However, in another PR it would be cool to switch go-plugin to use Unix sockets instead of TCP by default if that's possible now.